source/assets/javascripts/locastyle/_table.js

```diff
@@ -0,0 +1,104 @@
+var locastyle = locastyle || {};
+
+locastyle.table = (function() {
+  'use strict';
+
+  var config = {
+    selector: {
+      module: '[data-ls-module=table]',
+      mainCheck: '.ls-main-check',
+      selected: '.ls-selected'
+    },
+    classes: {
+      selected: 'ls-selected'
+    },
+    events: {
+      click: 'click.ls'
+    }
+  };
+
+  function unbindCheckboxes() {
+    $(config.selector.module).find('input[type=checkbox]').each(function() {
+      $(this).off(config.events.click);
+    });
+  }
+
+  function checkAllRows(table, mainCheck) {
+    mainCheck.on(config.events.click, function() {
+      if ($(this).prop('checked')) {
+        table.find('tbody tr').not(config.selector.selected).find('input[type=checkbox]').trigger(config.events.click);
```

Comment:
And if someone don't use the **tbody tr**? You are tying your functionality to one specific html structure...